policy/src/main/java/org/apache/brooklyn/policy/location/CreateLocationPolicy.java

```diff
@@ -0,0 +1,288 @@
+    }
+
+    private final SensorEventListener<Boolean> lifecycleEventHandler = new SensorEventListener<Boolean>() {
+        @Override
+        public void onEvent(SensorEvent<Boolean> event) {
+            synchronized (mutex) {
+                Boolean status = event.getValue();
+                if (status) {
+                    if (created.compareAndSet(false, true)) {
+                        LOG.info("Creating new location {} of type {}", getLocationCatalogItemId(), getLocationType());
+
+                        ImmutableList.Builder<String> builder = ImmutableList.<String>builder().add(
+                                "brooklyn.catalog:",
+                                "  id: " + getLocationCatalogItemId(),
+                                "  version: " + version.incrementAndGet(),
+                                "  itemType: location",
+                                "  item:",
+                                "    type: " + getLocationType(),
+                                "    brooklyn.config:",
+                                "      displayName: " + getLocationDisplayName());
+
+                        for (Map.Entry<String, AttributeSensor<?>> entry : getLocationConfiguration().entrySet()) {
+                            AttributeSensor<?> sensor = getLocationConfiguration().get(entry.getKey());
+                            Object value = entity.sensors().get(sensor);
+                            builder.add("      " + entry.getKey() + ": " + value);
+                        }
+                        if (getLocationTags().size() > 0) {
+                            builder.add("      tags:");
+                            for (String tag : getLocationTags()) {
+                                builder.add("        - " + tag);
+                            }
+                        }
+                        String blueprint = Joiner.on("\n").join(builder.build());
+                        LOG.debug("Creating location {} from YAML\n{}", getLocationCatalogItemId(), blueprint);
+
+                        Iterable<? extends CatalogItem<?,?>> catalogItems = getManagementContext().getCatalog().addItems(blueprint, true);
+                        int items = Iterables.size(catalogItems);
+                        if (items > 1) {
+                            LOG.warn("Got {} catalog items for location {}, using first: {}",
+                                    new Object[] { items, getLocationCatalogItemId(), Iterables.transform(catalogItems, item -> item.getCatalogItemId()) });
+                        } else if (items == 0) {
+                            throw new IllegalStateException("No catalog items returned for location: " + getLocationCatalogItemId());
+                        }
+                        catalogItem = Iterables.get(catalogItems, 0);
+                    }
+                } else {
+                    if (created.compareAndSet(true, false) && catalogItem != null) {
```

Comment:
`created` and `catalogItem` fields won't survive rebind. Neither will the auto-incrementing version. Therefore this will likely misbehave on rebind.